components/director/internal/tenantfetcher/service.go

```diff
@@ -230,14 +226,8 @@ func (s Service) SyncTenants() error {
 }
 
 func (s Service) createTenants(ctx context.Context, currTenants map[string]string, eventsTenants []model.BusinessTenantMappingInput) error {
-	tenantsToCreate := make([]model.BusinessTenantMappingInput, 0)
+	tenantsToCreate := s.parents(currTenants, eventsTenants)
 	for _, eventTenant := range eventsTenants {
-		if _, ok := currTenants[eventTenant.ExternalTenant]; ok {
-			continue
-		}
-		if len(eventTenant.Parent) > 0 {
-			eventTenant.Parent = currTenants[eventTenant.Parent]
-		}
```

Comment:
I think that here we should still do some checks:

```go
if _, ok := currTenants[eventTenant.ExternalTenant]; ok {
 	continue
 }
if len(eventTenant.Parent) > 0 {
        if parentInternalID, ok := currTenants[eventTenant.Parent]; ok {
 	        eventTenant.Parent = parentInternalID
        }
}
```

1. If the tenant already exists -> skip it
2. If the tenant has parent and the parent already exists -> convert the ID, since the common repo logic expects already converted parent ID if the parent is not being created in the same request. https://github.com/kyma-incubator/compass/blob/main/components/director/internal/domain/tenant/service.go#L122

Reply:
I agree for point 2, but for point one it won't fix the GA labels (this was supposed to be part of the previous PR)